nvflare/fuel/flare_api/api_spec.py

```diff
@@ -205,6 +227,256 @@ def delete_job(self, job_id: str):
 
     @abstractmethod
     def get_system_info(self) -> SystemInfo:
+        """Get general info of the FLARE system"""
+        pass
+
+    def get_client_job_status(self, client_names: List[str] = None) -> List[dict]:
+        """Get job status info of specified FL clients
+
+        Args:
+            client_names: names of the clients to get status info
+
+        Returns: A list of jobs running on the clients. Each job is described by a dict of: id, app name and status.
+        If there are multiple jobs running on one client, the list contains one entry for each job for that client.
+        If no FL clients are connected or the server failed to communicate to them, this method returns None.
+
+        """
+        pass
+
+    def restart(self, target_type: str, client_names: Optional[List[str]] = None) -> dict:
```

Comment:
Since this is a "spec", or "abstract class".
It might be better if we add `@abstractmethod` to these method definitions
SO we enforced whoever inherits this class need to implement these methods.

Reply:
implemented as suggested